test/telemetry_metrics/reporter_test.exs

```diff
@@ -249,4 +249,35 @@ defmodule Membrane.TelemetryMetrics.Reporter.Test do
 
     Reporter.stop(reporter)
   end
+
+  test "Only one process is created for metrics from the same process" do
+    metrics = [
+      Telemetry.Metrics.counter("counter", event_name: [:event])
+    ]
+
+    {:ok, reporter} = Reporter.start_link(metrics)
+
+    Membrane.TelemetryMetrics.register([:event], id: 0)
+
+    Membrane.TelemetryMetrics.execute([:event], %{}, %{}, id: 0)
+
+    process_num =
+      Process.list()
+      |> Enum.count()
+
+    Membrane.TelemetryMetrics.register([:event], id: 1)
+
+    Membrane.TelemetryMetrics.execute([:event], %{}, %{}, id: 1)
+
+    assert ^process_num =
+             Process.list()
+             |> Enum.count()
```

Comment:
Maybe using `Process.info(self(), :monitored_by | :monitors` (take a look at https://www.erlang.org/doc/man/erlang.html#process_info-2) would be a better idea? This test might be interrupted by some leftovers from other tests